### PR TITLE
Install content packs one by one in the XSIAM nightly

### PR DESCRIPTION
**Pull request by daryakoval**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues

https://jira-hq.paloaltonetworks.local/browse/CIAC-5229

fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-5426

Nightly: https://code.pan.run/xsoar/content/-/pipelines/4474951
Took 30 minutes to pass.

### CODE REVIEW

File: Tests/Marketplace/search_and_install_packs.py [hunk 1]
```diff
@@ -409,7 +409,10 @@ def search_pack_and_its_dependencies(client: demisto_client,
                                      pack_id: str,
                                      packs_to_install: list,
                                      installation_request_body: list,
-                                     lock: Lock):
+                                     lock: Lock,
+                                     one_pack_and_its_dependencies_in_batch: bool = False,
+                                     batch_packs_install_request_body: list = None
```
Comment (tkatzir) [inline Tests/Marketplace/search_and_install_packs.py:414]:
```suggestion
                                     batch_packs_install_request_body: list = None,
```

File: Tests/Marketplace/search_and_install_packs.py [hunk 2]
```diff
@@ -634,13 +645,16 @@ def search_and_install_packs_and_their_dependencies_private(test_pack_path: str,
 
 
 def search_and_install_packs_and_their_dependencies(pack_ids: list,
-                                                    client: demisto_client, hostname: str = ''):
+                                                    client: demisto_client, hostname: str = '',
+                                                    install_packs_one_by_one=False):
```
Comment (tkatzir) [inline Tests/Marketplace/search_and_install_packs.py:649]: Same suggestion as above - Change to an `int` parameter of the batch_size.

File: Tests/configure_and_test_integration_instances.py [hunk 3]
```diff
@@ -337,7 +338,8 @@ def install_packs(self, pack_ids=None):
         for server in self.servers:
             try:
                 hostname = self.cloud_machine if self.is_cloud else ''
-                _, flag = search_and_install_packs_and_their_dependencies(pack_ids, server.client, hostname)
+                _, flag = search_and_install_packs_and_their_dependencies(pack_ids, server.client, hostname,
+                                                                          install_packs_one_by_one)
```
Comment (tkatzir) [inline Tests/configure_and_test_integration_instances.py:342]:
```suggestion
                                                                          install_packs_one_by_one,
                                                                         )
```

**Review by tkatzir — APPROVED** [review]
Approved, with comments.

Also - If you implement batch_size, then you could extract the batches calculation to a method which you could unit-test.

File: Tests/Marketplace/search_and_install_packs.py (hunk 2 above)
Comment (daryakoval) [reply Tests/Marketplace/search_and_install_packs.py:649]: I did it before, but decided it is not so good practice, https://github.com/demisto/content/pull/23902/commits/d76e115ad2d8c8ce8bdd224007aa22135ab29381
 
 I can explain why on zoom call :)

File: Tests/Marketplace/search_and_install_packs.py [hunk 4]
```diff
@@ -409,7 +409,10 @@ def search_pack_and_its_dependencies(client: demisto_client,
                                      pack_id: str,
                                      packs_to_install: list,
                                      installation_request_body: list,
-                                     lock: Lock):
+                                     lock: Lock,
+                                     one_pack_and_its_dependencies_in_batch: bool = False,
```
[the remark this answers is not in the document]
Comment (daryakoval) [reply Tests/Marketplace/search_and_install_packs.py:413]: See this commit in this pr: https://github.com/demisto/content/pull/23902/commits/d76e115ad2d8c8ce8bdd224007aa22135ab29381

Let's discuss it.

**Pull request merged**